### PR TITLE
refactor: add algorithm option to SmoothQuantModifier, deprecate LogarithmicEqualizationModifier

### LINKED ISSUE #2554: [Refactor] Remove `LogarithmicEqualizationModifier`
LogarithmicEqualizationModifier should just be an option of SmoothQuantModifier, not an entirely separate modifier.

### PR DESCRIPTION
**Pull request by wiliyam**
## Summary
Closes #2554

`LogarithmicEqualizationModifier` is just `SmoothQuantModifier` with a different scale calculation. Instead of maintaining two separate classes, this adds an `algorithm` parameter to `SmoothQuantModifier`.

## Changes

**`SmoothQuantModifier`** — added `algorithm: str = "smoothquant"` field:
- `"smoothquant"` (default): existing behaviour unchanged
- `"log_equalization"`: uses the log2 scale formula from the Logarithmic Equalization paper (https://arxiv.org/abs/2308.15987)

**`LogarithmicEqualizationModifier`** — now a deprecated alias:
- Emits `DeprecationWarning` on instantiation
- Delegates to `SmoothQuantModifier(algorithm='log_equalization')`
- Will be removed in a future release

## Migration

```yaml
# Before
LogarithmicEqualizationModifier:
  mappings: [...]

# After
SmoothQuantModifier:
  algorithm: log_equalization
  mappings: [...]
```

### CODE REVIEW

**Comment by wiliyam** [comment]
PR is ready for review! Signed commits with DCO.

**Comment by wiliyam** [comment]
Could a maintainer add the `ready` label? The CI check requires it and I don't have permission to add labels as an external contributor. PR has DCO sign-off and is ready for review. Thanks!

**Comment by wiliyam** [comment]
Addressed Gemini's feedback:
1. **Validation** — changed `algorithm: str` to `algorithm: Literal["smoothquant", "log_equalization"]` so Pydantic rejects typos at instantiation time
2. **Redundant weight computation** — `log_equalization` now returns early before computing `weight_scales` (it doesn't need them), avoiding unnecessary per-layer weight ops and the potential behavioral regression from the `torch.where` fallback

Could a maintainer also add the `ready` label when convenient? 🙏

**Comment by wiliyam** [comment]
Fixed the `make style` / `make quality` failure — one line in the deprecation docstring was 89 chars (limit is 88). Reformatted with ruff, all checks pass now.

**Comment by wiliyam** [comment]
Done @kylesayrs — removed both test files:\n- `tests/llmcompressor/modifiers/logarithmic_equalization/test_base.py`\n- `tests/llmcompressor/pytorch/modifiers/logarithmic_equalization/test_pytorch.py`

**Comment by wiliyam** [comment]
Thanks @kylesayrs! The test files are already removed in this PR (both show `status: removed` in the diff). Ready for merge whenever you are 🙏

**Review by brian-dellabetta — APPROVED** [review]
Thanks for the contribution! LGTM

**Comment by wiliyam** [comment]
Thank you @kylesayrs for the review and merge! 🙏
